tensorflow/compiler/xla/service/llvm_ir/llvm_target_features.h

```diff
@@ -0,0 +1,112 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#ifndef TENSORFLOW_COMPILER_XLA_SERVICE_LLVM_IR_LLVM_TARGET_FEATURES_H_
+#define TENSORFLOW_COMPILER_XLA_SERVICE_LLVM_IR_LLVM_TARGET_FEATURES_H_
+
+#include <string>
+#include "llvm/IR/Intrinsics.h"
+
+namespace xla {
+namespace llvm_ir {
+// Enmerations to get target specific intrinsics or function calls
+typedef enum  TargetIntrinsicEnum {
+  kSHFL_DOWN_F32 = 0,
+  kSHFL_DOWN_I32 = 1,
+  kTHREAD_ID_X = 2,
+  kTHREAD_ID_Y = 3,
+  kTHREAD_ID_Z = 4,
+  kBLOCK_ID_X = 5,
+  kBLOCK_ID_Y = 6,
+  kBLOCK_ID_Z = 7,
+  kBARRIER_ID = 8,
+  kLAST = 9
+} TargetIntrinsicEnum;
+
+// Abstract interface for classes providing information about the gpu target
+// we're compiling for.
+class LLVMTargetFeatures {
+  const llvm::Intrinsic::ID* tgt_intrinsic;
```

Comment:
`tgt_intrinsic` is not a good name for this variable; it's actually an array of intrinsics, not just one intrinsic.

Perhaps `intrinsics` or `intrinsics_array`?